src/utils/file_utils.py

```diff
@@ -0,0 +1,188 @@
+    logger.info(f"Normalized file path for hashing: {normalized_path}")
+
+    # Convert the normalized path to bytes and hash it
+    file_path_bytes = normalized_path.encode('utf-8')
+    hash_object = hashlib.sha256()
+    hash_object.update(file_path_bytes)
+    
+    # Generate the hexadecimal digest
+    unique_identifier = hash_object.hexdigest()
+    
+    logger.info(f"HASH in GET UNIQUE ID for: {normalized_path} HASH={unique_identifier}")
+    
+    return unique_identifier
+
+def generate_translated_filename(original_filepath: str | Path, language_code: str, root_dir: Path) -> str:
+    """
+    Generate a filename for a translated file, including a unique hash and language code.
+
+    Note:
+    If the file path and the file name are identical, the same hash will be generated.
+    This is because the hash is based on the entire file path.
+
+    Args:
+        original_filepath (str): The original file path.
+        language_code (str): The language code for the translation (e.g., 'en', 'fr').
+
+    Returns:
+        str: The translated file's new filename.
+    """
+
+    original_filepath = Path(original_filepath)
+
+    # Extract original file components
+    original_filename, file_ext = get_filename_and_extension(original_filepath)
+
+    # Extract filename and extension
+    unique_hash = get_unique_id(str(original_filepath), root_dir)
+
+    # Generate the new filename with the unique hash and language code
+    new_filename = f"{original_filename}.{unique_hash}.{language_code}{file_ext}"
+
+    return new_filename
+
+def get_filename_and_extension(file_path: str | Path) -> tuple[str, str]:
+    """
+    Extract the filename without extension and the file extension from the given file path.
+
+    Args:
+        file_path (str | Path): The file path from which to extract the filename and extension.
+
+    Returns:
+        tuple[str, str]: A tuple containing the filename without extension and the file extension.
+    """
+    # Ensure the file_path is a string or Path object
+    file_path = str(file_path)  # Convert to string if it's a Path object
+
+    # Extract the base name (filename with extension) and split it into name and extension
+    original_filename, file_ext = os.path.splitext(os.path.basename(file_path))
+
+    # Return the filename without extension and the file extension in lowercase
+    return original_filename, file_ext.lower()
+
+def filter_files(directory: str | Path) -> list:
+    """
+    Filter and return only the files in the given directory, excluding directories.
+
+    Args:
+        directory (str | Path): The directory path to search for files.
+
+    Returns:
+        list: A list of Path objects representing only the files (excluding directories).
+    """
+    directory = Path(directory)
+    return [file for file in directory.glob('**/*') if file.is_file()]
+
+def get_translation_destination(file_path: Path, root_dir: Path, folder_type: str) -> Path:
```

Comment:
Planning to remove the `get_translation_destination` function as it's no longer necessary. I believe it would be better to develop this functionality later when it's required in the future.